recipes-bsp/u-boot/u-boot-variscite/0003-HMX-custom-imx8mp_var_dart.patch

```diff
@@ -0,0 +1,255 @@
++		init_uart_clk(2);
+ 	}
+ 	else if (board_id == BOARD_ID_SOM) {
+ 		imx_iomux_v3_setup_multiple_pads(uart_pads_som,
+@@ -112,8 +120,55 @@ int board_early_init_f(void)
+ }
+ 
+ #ifdef CONFIG_OF_BOARD_SETUP
++/* Set serial number and part number from environment to device tree blob chosen.
++		serial-number = "23078888";
++		part-number = "hmp00xx"; 
++*/
+ int ft_board_setup(void *blob, struct bd_info *bd)
+ {
++#if defined(CONFIG_OF_LIBFDT)
++	char *serial;
++	char *part;
++	int offset;
++	int err = 1;
++	const char *path;
++
++	path = "/chosen";
++	offset = fdt_path_offset(blob, path);
++	if (offset < 0) {
++		debug("Node %s not found.\n", path);
++		return err;
++	}
++
++	debug("ft_board_setup: config FDT\n");
++
++	serial = env_get("serial");
++	if (serial) {
++		err = fdt_setprop(blob, offset, "serial-number", serial,
++				  strlen(serial) + 1);
++
++		if (err < 0) {
++			printf("WARNING: could not set serial-number %s.\n",
++			       fdt_strerror(err));
++			return err;
++		}
++	} else {
++		printf("WARNING: could not read serial-number from env list.\n");
++	}
++	part = env_get("part_nr");
++	if (part) {
++		err = fdt_setprop(blob, offset, "part-number", part,
++				  strlen(part) + 1);
++
++		if (err < 0) {
++			printf("WARNING: could not set part-number %s.\n",
++			       fdt_strerror(err));
++			return err;
++		}
++	} else {
++		printf("WARNING: could not read part-number from env list.\n");
++	}
++#endif
+ #ifdef CONFIG_IMX8M_DRAM_INLINE_ECC
+ 	int rc;
+ 	phys_addr_t ecc0_start = 0xb0000000;
+@@ -337,6 +392,9 @@ int board_init(void)
+ 	init_usb_clk();
+ #endif
+ 
++//does not compile. #if defined(CONFIG_HW_WATCHDOG) || defined(CONFIG_WATCHDOG)
++//	hw_watchdog_init();
++//#endif
+ 	return 0;
+ }
+ 
+@@ -350,6 +408,9 @@ int board_late_init(void)
+ 	struct var_carrier_eeprom carrier_eeprom;
+ 	char carrier_rev[CARRIER_REV_LEN] = {0};
+ 
++	uchar eeprom_data [0xffff];
++	int err = 1;
++
+ #ifdef CONFIG_ENV_IS_IN_MMC
+ 	board_late_mmc_env_init();
+ #endif
+@@ -382,6 +443,28 @@ int board_late_init(void)
+ 	var_setup_mac(ep);
+ 	var_eeprom_print_prod_info(ep);
+ 
++	/*HMX detect usb boot button for upgrade mmc from usb memory*/
++	imx_iomux_v3_setup_multiple_pads(usb_upgrade_button_pad,
++				ARRAY_SIZE(usb_upgrade_button_pad));
++	gpio_request(USB_UPGRADE_BUTTON_GPIO, "USB_UPGRADE_BUTTON");
++	gpio_direction_input(USB_UPGRADE_BUTTON_GPIO);
++	if (gpio_get_value(USB_UPGRADE_BUTTON_GPIO) == 0 )
++	{
++		env_set("usb_flash_wanted", "true");
++	}
++	gpio_free(USB_UPGRADE_BUTTON_GPIO);
++	/*HMX get env from eeprom serial number, partnumber and mac adress(eth2addr, eth3addr) and more*/
++	err = get_environment_data_from_eeprom(eeprom_data);
++	if ( err != 0)
++	{
++		printf("Get environment data from eeprom failed, error:%x\n", err);
++	} else {
++		err = set_environment_data_from_eeprom((char *)eeprom_data);
++		if (err != 0)
++		{
++			printf("Set environment data to  environment area failed, error:%x\n", err);
++		}
++	}
```

Comment:
Just coding style fixes:

```suggestion
+	if (gpio_get_value(USB_UPGRADE_BUTTON_GPIO) == 0) {
+		env_set("usb_flash_wanted", "true");
+	}
+	gpio_free(USB_UPGRADE_BUTTON_GPIO);
+	/* HMX get env from eeprom serial number, partnumber and mac adress(eth2addr, eth3addr) and more */
+	err = get_environment_data_from_eeprom(eeprom_data);
+	if (err != 0) {
+		printf("Get environment data from eeprom failed, error:%x\n", err);
+	} else {
+		err = set_environment_data_from_eeprom((char *)eeprom_data);
+		if (err != 0) {
+			printf("Set environment data to environment area failed, error: %x\n", err);
+		}
+	}
```

Reply:
This is a patch so the line changes makes a compile issue. in this case three lines was removed and "_@@ -382,6 +443,28 @@ int board_late_init(void)_" should be "_@@ -382,6 +443,25 @@ int board_late_init(void)_"